sql/sql_servers.cc

```diff
@@ -112,6 +112,54 @@ static void init_servers_cache_psi_keys(void)
 }
 #endif /* HAVE_PSI_INTERFACE */
 
+
+bool trt_table_type_is_correct(TABLE *t)
+{
+  if (t->file->ht->db_type != DB_TYPE_INNODB)
+    return false;
+
+  if (t->s->fields != 5)
+    return false;
+
+  if (t->field[0]->type() != MYSQL_TYPE_LONGLONG)
+    return false;
+
+  if (t->field[1]->type() != MYSQL_TYPE_LONGLONG)
+    return false;
+
+  if (t->field[2]->type() != MYSQL_TYPE_TIMESTAMP)
+    return false;
+
+  if (t->field[3]->type() != MYSQL_TYPE_TIMESTAMP)
+    return false;
+
+  if (t->field[4]->type() != MYSQL_TYPE_TINY)
+    return false;
+
+  if (!t->key_info || !t->key_info->key_part)
+    return false;
+
+  if (strcmp(t->key_info->key_part->field->field_name.str, "transaction_id"))
+    return false;
+
+  return true;
+}
+
+bool trt_table_correct(THD *thd, bool &correct)
```

Comment:
Move to `TR_table::check()`. I don't see the reason why the code on opening the table should be duplicated.